#! /usr/bin/env python

# Circle tracker dynamic parameter configuration file

PACKAGE='circle_tracker'

from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

# Name, Type, Level, Description, Default, Min, Max

gen.add("high_canny_threshold", double_t, 0, "Canny edge detector threshold", 120, 0, 255)

gen.add("point_count_threshold", int_t, 0, "Pount count threshold", 2, 6, 20)

gen.add("area_threshold", double_t, 0, "Area (pixels) threshold", 50, 10, 500)

gen.add("circle_threshold", double_t, 0, "Circularity threshold", 0.75, 0, 1)

gen.add("blur_sigma", double_t, 0, "Gaussian blur sigma (< 0.5 disables)", 0, 0, 2.0)

exit(gen.generate(PACKAGE, "circle_tracker", "CircleTracker"))
